src/main.cpp

```diff
@@ -1367,23 +1301,118 @@ int main(int Argc, char *Argv[]) {
+    return ArgsParseResult::success;
+}
+
+int main(int Argc, char *Argv[]) {
+    std::vector<char *> argv;
+    lGetAllArgs(Argc, Argv, argv);
+    int argc = argv.size();
+#ifdef ISPC_HOST_IS_WINDOWS
+    // While ispc doesn't load any libraries explicitly using LoadLibrary API (or alternatives), it uses vcruntime that
+    // loads vcruntime140.dll and msvcp140.dll. Moreover LLVM loads dbghelp.dll.
+    // There is no way to modify DLL search order for vcruntime140.dll and msvcp140.dll but we
+    // can prevent searching in CWD while loading dbghelp.dll.
+    // So before initiating any LLVM call, remove CWD from the search path to reduce the risk of DLL injection
+    // when Safe DLL search mode is OFF.
+    // https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order
+    SetDllDirectory("");
+#endif
+    llvm::sys::AddSignalHandler(lSignal, nullptr);
+    // initialize available LLVM targets
+#ifdef ISPC_X86_ENABLED
+    LLVMInitializeX86TargetInfo();
+    LLVMInitializeX86Target();
+    LLVMInitializeX86AsmPrinter();
+    LLVMInitializeX86AsmParser();
+    LLVMInitializeX86Disassembler();
+    LLVMInitializeX86TargetMC();
+#endif
+
+#ifdef ISPC_ARM_ENABLED
+    LLVMInitializeARMTargetInfo();
+    LLVMInitializeARMTarget();
+    LLVMInitializeARMAsmPrinter();
+    LLVMInitializeARMAsmParser();
+    LLVMInitializeARMDisassembler();
+    LLVMInitializeARMTargetMC();
+
+    LLVMInitializeAArch64TargetInfo();
+    LLVMInitializeAArch64Target();
+    LLVMInitializeAArch64AsmPrinter();
+    LLVMInitializeAArch64AsmParser();
+    LLVMInitializeAArch64Disassembler();
+    LLVMInitializeAArch64TargetMC();
+#endif
+
+#ifdef ISPC_WASM_ENABLED
+    LLVMInitializeWebAssemblyAsmParser();
+    LLVMInitializeWebAssemblyAsmPrinter();
+    LLVMInitializeWebAssemblyDisassembler();
+    LLVMInitializeWebAssemblyTarget();
+    LLVMInitializeWebAssemblyTargetInfo();
+    LLVMInitializeWebAssemblyTargetMC();
+#endif
+    char *file = nullptr;
+    const char *headerFileName = nullptr;
+    const char *nanobindWrapperFileName = nullptr;
+    const char *outFileName = nullptr;
+    const char *depsFileName = nullptr;
+    const char *depsTargetName = nullptr;
+    const char *hostStubFileName = nullptr;
+    const char *devStubFileName = nullptr;
+
+    std::vector<std::string> linkFileNames;
+    // Initiailize globals early so that we can set various option values
+    // as we're parsing below
+    g = new Globals;
+
+    Module::OutputType ot = Module::Object;
+    Module::OutputFlags flags;
+    Arch arch = Arch::none;
+    std::vector<ISPCTarget> targets;
+    const char *cpu = nullptr;
+
+    std::string ISPCAbsPath = llvm::sys::fs::getMainExecutable(argv[0], (void *)(intptr_t)main);
+    initializeBinaryType(ISPCAbsPath.c_str());
+
+    // Parse command line optionsAdd commentMore actions
+    bool isLinkMode = false;
+    ArgsParseResult parseResult = ParseCommandLineArgs(
```

Comment:
What if `ParseComandLineArgs` directly fills the `Module::Output` in the first place:

```c++
    ArgsParseResult parseResult = ParseCommandLineArgs(argc, argv.data(), file, arch, cpu, targets, output, linkFileNames, isLinkMode);
```

Reply:
I planned to do it as part of another refactoring PR but it makes sense here.